src/Umbraco.Core/Services/ContentValidationServiceBase.cs

```diff
@@ -34,27 +35,52 @@ protected async Task<ContentValidationResult> HandlePropertiesValidationAsync(
             .ToArray();
         IPropertyType[] variantPropertyTypes = contentTypePropertyTypes.Except(invariantPropertyTypes).ToArray();
 
+        var cultures = culturesToValidate?.WhereNotNull().Except(["*"]).ToArray();
+        if (cultures?.Any() is not true)
+        {
+            cultures = await GetCultureCodes();
+        }
+
+        // we don't have any managed segments, so we have to make do with the ones passed in the model
+        var segments = contentEditingModelBase.Variants
+            .Where(variant => variant.Culture is null || cultures.Contains(variant.Culture))
+            .DistinctBy(variant => variant.Segment).Select(variant => variant.Segment)
+            .WhereNotNull()
+            .ToArray();
+
         foreach (IPropertyType propertyType in invariantPropertyTypes)
         {
-            validationErrors.AddRange(ValidateProperty(contentEditingModelBase, propertyType, null, null));
+            var validationContext = new PropertyValidationContext
+            {
+                Culture = null, Segment = null, CulturesBeingValidated = cultures, SegmentsBeingValidated = segments
+            };
+
+            PropertyValueModel? propertyValueModel = contentEditingModelBase.InvariantProperties.FirstOrDefault(pv => pv.Alias == propertyType.Alias);
```

Comment:
A bit pedantic, but i prefer full names over abbreviations (and we're doing it above with `variant`)
```suggestion
            PropertyValueModel? propertyValueModel = contentEditingModelBase.InvariantProperties.FirstOrDefault(propertyValue => propertyValue.Alias == propertyType.Alias);
```